tests/model_formsets/tests.py

```diff
@@ -2432,3 +2433,55 @@ class ModelFormWithDefaultRenderer(ModelForm):
             formset.empty_form.renderer, ModelFormWithDefaultRenderer.default_renderer
         )
         self.assertIsInstance(formset.renderer, DjangoTemplates)
+
+
+class BaseInlineFormSetTest(TestCase):
+
+    def test_inline_formset_with_custom_auto_id(self):
+        """
+        The auto_id attribute can be set to a custom value.
+        """
+
+        class PoemModelForm(ModelForm):
+            pass
+
+        poet = Poet.objects.create(name="test")
+        poet.poem_set.create(name="first test poem")
+        poet.poem_set.create(name="second test poem")
+        PoemInlineFormSet = inlineformset_factory(
+            Poet,
+            Poem,
+            form=PoemModelForm,
+            formset=BaseInlineFormSet,
+            fields=("name",),
+            extra=0,
+        )
+
+        # The auto_id attribute can be set to a string template using keyword argument.
+
+        inlineformset = PoemInlineFormSet(None, instance=poet, auto_id="id_custom_%s")
+        self.assertEqual(inlineformset.forms[0].auto_id, "id_custom_%s")
+
+        # Using positional argument
+        inlineformset = PoemInlineFormSet(
+            None, None, poet, False, None, None, "id_custom_%s"
```

Comment:
```suggestion
            None, None, poet, False, None, None, auto_id="id_custom_%s"
```
In the test, we would need to pass the `auto_id` not as a named argument.